x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/cohere/CohereAccount.java

```diff
@@ -7,15 +7,26 @@
 
 package org.elasticsearch.xpack.inference.external.cohere;
 
+import org.elasticsearch.common.CheckedSupplier;
 import org.elasticsearch.common.settings.SecureString;
-import org.elasticsearch.core.Nullable;
+import org.elasticsearch.xpack.inference.services.cohere.CohereModel;
 
 import java.net.URI;
+import java.net.URISyntaxException;
 import java.util.Objects;
 
-public record CohereAccount(@Nullable URI url, SecureString apiKey) {
+import static org.elasticsearch.xpack.inference.external.request.RequestUtils.buildUri;
+
+public record CohereAccount(URI uri, SecureString apiKey) {
```

Comment:
I'm a bit confused by what you mean. Are you saying that the uri can be null, but only for testing?

Reply:
I had a misspelling in my comment 😅 , not sure if that helps clear it up but what I meant to say was: `I'm now relying...`.

Another side not, this change actually isn't necessary but I left it in so all the services are taking the same approach now. Originally I thought the rate limiting for cohere would need the URL but I don't think we do. But anyway. Prior to this PR we were initializing the `CohereAccount` with the immediate results from the service settings. The service settings allow a `url` field to be passed to override the actual cohere url. So technically the user _could_ initialize an inference endpoint with a custom url if they wanted. The reason I added that originally was to help with testing so I could spin up a mock server and pass the url through the various parts of the cohere logic. In the normal production use case, the `url` field will be null (because the user won't include it) and then the account here will contain `null`. Then when we go to create a cohere request we check the account first, if the uri is null we use the default one. All that being said, what this change is doing is moving the logic to do the null check inside this `of()` call so the account will always have a non-null URI.

Reply:
nice, thanks!